community/values/src/test/java/org/neo4j/values/storable/NanosecondTruncationTest.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.values.storable;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.ZoneId;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.neo4j.values.AnyValue;
+import org.neo4j.values.virtual.VirtualValues;
+
+import static java.time.ZoneOffset.UTC;
+import static java.time.temporal.ChronoField.MICRO_OF_SECOND;
+import static java.time.temporal.ChronoField.NANO_OF_SECOND;
+import static java.time.temporal.ChronoUnit.NANOS;
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.neo4j.values.storable.DateTimeValue.datetime;
+import static org.neo4j.values.storable.DurationValue.duration;
+import static org.neo4j.values.storable.LocalDateTimeValue.localDateTime;
+import static org.neo4j.values.storable.LocalTimeValue.localTime;
+import static org.neo4j.values.storable.TimeValue.time;
+import static org.neo4j.values.utils.AnyValueTestUtil.assertEqual;
+import static org.neo4j.values.utils.AnyValueTestUtil.assertNotEqual;
+
+public class NanosecondTruncationTest
+{
+    static final Supplier<ZoneId> inUTC = () -> UTC;
+
+    private int prevPrecision;
+
+    @Before
+    public void setup()
+    {
+        prevPrecision = Values.getNanoPrecision();
+        Values.setNanoPrecision( 6 );
+    }
+
+    @After
+    public void tearDown()
+    {
+        Values.setNanoPrecision( prevPrecision );
```

Comment:
This should not be possible

Reply:
But unfortunately required for testing, with the current solution. The JVM is not restarted between test runs and different tests expect different values for the nano precision.